mmdet/datasets/coco.py

```diff
@@ -87,39 +87,27 @@ def get_cat_ids(self, idx):
     def _filter_imgs(self, min_size=32):
         """Filter images too small or without ground truths."""
         valid_inds = []
+        # obtain images that contain annotation
         ids_with_ann = set(_['image_id'] for _ in self.coco.anns.values())
+        # obtain images that contain annotations of the required categories
```

Comment:
It's not necessary to do the filtering here. 
We can directly do that in `load_annotations`,
`self.cat_ids = self.coco.get_cat_ids(cat_names=self.CLASSES)`
`self.img_ids = self.coco.get_img_ids(cat_ids=self.cat_ids)`

Reply:
self.coco.get_img_ids will only return images that contain all the categories specified by the `cat_ids`.